### PR TITLE
Handle the cases where < 3 people competed in 3x3x3 final round.

### LINKED ISSUE #1607: Posting results announcement fails when < 3 people competed in 3x3x3 final round
At [Korean Spring Picnic 2017](https://www.worldcubeassociation.org/competitions/KoreanSpringPicnic2017), only 2 people competed in 3x3 final round, which triggered [this code](https://github.com/thewca/worldcubeassociation.org/blob/ef4e57980de3b73d83abf63a179ec52212704ffc/WcaOnRails/app/controllers/competitions_controller.rb#L200).

@SAuroux doesn't really care how this is handled, so we just need to some somthing sensible here (the most reasonable feeling thing to me is to only announce 1 person if 1 person competed, and only 2 people if 2 people competed).

### PR DESCRIPTION
**Pull request by jfly**
This fixes #1607.

Added some tests for this code, too!

### CODE REVIEW

File: WcaOnRails/app/controllers/competitions_controller.rb [hunk 1]
```diff
@@ -196,19 +196,21 @@ def post_results
...
 
         body = "[#{top333.first.personName}](#{person_url top333.first.personId})"
         body += " won the [#{comp.name}](#{competition_url(comp)})"
-        body += " with an average of #{top333.first.to_s :average} seconds."
+        body += " with an average of #{top333.first.to_s :average} seconds"
 
-        body += " [#{top333.second.personName}](#{person_url top333.second.personId}) finished second (#{top333.second.to_s :average})"
+        if top333.length >= 2
+          body += ". [#{top333.second.personName}](#{person_url top333.second.personId}) finished second (#{top333.second.to_s :average})"
+          if top333.length >= 3
```
Comment (jonatanklosko) [inline WcaOnRails/app/controllers/competitions_controller.rb]: Super nit: there's no way to be more than 3 of those, right? =)
Maybe `> 1` and `> 2`?

File: WcaOnRails/spec/controllers/competitions_controller_spec.rb [hunk 2]
```diff
@@ -566,7 +566,71 @@
...
+      it "handles no 3x3x3 event" do
+        get :post_results, params: { id: competition }
+        post = assigns(:post)
+        expect(post.title).to eq "Results of #{competition.name}, in #{competition.cityName}, #{competition.countryId} posted"
+        expect(post.body).to eq "Results of the [#{competition.name}](#{competition_url(competition)}) are now available.\n\n"
+      end
+
+      context "winners announcement" do
+        let(:add_result) do
+          lambda do |pos, name|
```
Comment (jonatanklosko) [inline WcaOnRails/spec/controllers/competitions_controller_spec.rb]: I believe this [can be a normal method](https://github.com/thewca/worldcubeassociation.org/blob/c2b39b5e984379ef9c7576be71f92607c57d4201/WcaOnRails/spec/lib/auxiliary_data_computation_spec.rb#L8-L17).

File: WcaOnRails/app/controllers/competitions_controller.rb (hunk 1 above)
Comment (jfly) [inline WcaOnRails/app/controllers/competitions_controller.rb]: Haha, I guess that's a clever way around this problem. Sure thing =)

File: WcaOnRails/spec/controllers/competitions_controller_spec.rb (hunk 2 above)
Comment (jfly) [inline WcaOnRails/spec/controllers/competitions_controller_spec.rb]: Ah, ok! Will do.

**Comment by jfly** [comment]
I actually forgot about controller specs being deprecated, thanks for the reminder. I don't see the benefit of porting our controller specs to request specs, but let's do keep an eye out and try to avoid creating new controller spec files.

**Pull request merged**